### ISSUE TITLE
License filter is no longer available, please return this necessary feature

### ISSUE THREAD

**Issue by Fishvap** [modrinth/knossos#828]
### Is your suggested enhancement related to a problem? Please describe.

I'd love to ask on the Modrinth Discord but I was banned a long time ago and I've got no clue how to appeal. I use Modrinth actively and I was banned purely because of the fact that Modrinth is heavily associated with Quilt, which shouldn't really be the case.

I've been trying to seek the License filter but it appears to have suddenly vanished without reason.

### Describe the solution you'd like

The filter is no longer available, please return the License filter. Open-source is vague and doesn't determine whether or not it's ARR + source code publicly available, or the license is open-source + source code publicly available. This has also made it harder to filter for mods that may have a custom license that may not be included within Modrinth.

### Describe alternatives you've considered

Reverse #817 so that the license filter returns.

**Comment by Fishvap** [comment]
Wrong repo, sorry

**Comment by Fishvap** [comment]
Actually I don't know if this is the right repo or not

**Comment by Fishvap** [comment]
It appears that #817 is the cause of the removal, as it removes the License category and replaces it with a really bad "Open source" category which doesn't allow you to determine the licenses. This feels like stripping away functionality that was legitimately useful.

**Comment by triphora** [comment]
Not planned. We did preliminary surveys beforehand (e.g. [here](https://twitter.com/modrinth/status/1540444402367201280)) and we were unable to find anyone who uses it. The new filter returns projects which have an OSI-approved license. I remember someone asking for a second filter that said "visible source", which would be easy to do, but an issue hasn't been opened for it yet and it will be easily forgotten if one isn't opened.

**Comment by Fishvap** [comment]
I'm really unhappy with this change, as I personally use it and find it very useful for modpacks. I don't see a huge reason for its removal, considering it must not be a difficult feature to maintain?

I'd appreciate if the License filter was returned until a sufficient alternative is made, because this is just crippling the site without a legitimate reason. The preferable alternative is to make the checkbox remove the licenses which may not fall under the Open Source license, such as ARR and Custom, and add these to a separate "blacklisted tags" below the "whitelisted tags" which indicates which licenses won't appear.

**Comment by Fishvap** [comment]
Also, Twitter is not a good way of doing surveys, especially since you did not make a poll. If you would like to poll for people's opinions, I would recommend at least a poll, or alternatively, ask for people's opinions through the site (however you may consider to do that, whether it be an external or an internal poll).

**Comment by triphora** [comment]
> and find it very useful for modpacks

You can use any Modrinth mod in Modrinth modpacks.

> considering it must not be a difficult feature to maintain

It cannot really exist as it did in its previous state after modrinth/labrinth#449. It brought major positive changes to creators using licenses other than the dozen or so we specified, in exchange for this relatively minor tradeoff. We were unable to find any legitimate use cases for filtering for specific licenses that was not covered by the open source checkbox.

> An additional point to make: "Open Source" does not include ARR/Custom which may have its source available to users for them to modify and distribute, but may not fall under the same restrictions as other licenses.

Intentional. With the information we have, it is impossible to see if custom licenses allow such things, and even if we had such a checkbox (e.g. "Allow distribution"), we would have no way of checking if the way they worded it in their license is legally enforceable. If authors really wanted their project to be open source, they would not be using a crayon license or ARR.

**Comment by Lancelotbronner** [comment]
I too would like to see this feature returned as I have been using it.

Open source and libre software are completely different and there is no way to make that distinction right now.

Also not any mod is available for modrinth modpacks as some authors will specifically request certain things which will be reflected by their license.

I was also not aware of a "poll" about people's opinions, and you have proof right here that people are using it since there are requests for you to bring it back.

In short, I think the new filter is too vague and that's why I believe a license filter should be returned. Those two ways to filter licenses can absolutely coexist.

**Comment by triphora** [comment]
> Also not any mod is available for modrinth modpacks as some authors will specifically request certain things which will be reflected by their license.

By uploading to Modrinth, authors give specific rights to Modrinth, including the rights to "use, reproduce, distribute copies, prepare derivative works of, or display Content in connection with Modrinth in any medium and for any purpose (including commercial purposes)."[^1] Modrinth extends this permission to Modrinth modpack authors, and any Modrinth project can be included in a Modrinth modpack. Ordinarily it would be against our content rules, but we specifically grant this permission to modpacks.[^2]

> Open source and libre software are completely different and there is no way to make that distinction right now.

Is there? We use the definition of open source as published by [the OSI](https://opensource.org/osd) for the purpose of this filter.[^3] If you simply want to see visible-source mods, that can be added as a separate filter - but again, make an issue for that.

> and you have proof right here that people are using it since there are requests for you to bring it back.

Two requests out of the multiple hundreds of thousands of unique visitors we have per day. It's impractical and impossible to honor every single request made by this small of a part of Modrinth's userbase.

> Those two ways to filter licenses can absolutely coexist.

Not really on the technical side of things. Unless you want the license filter to be 503 entries[^4] long, not include custom/ARR, and lag painfully slowly because vue-multiselect is terrible.

[^1]: https://modrinth.com/legal/terms#content
[^2]: https://modrinth.com/legal/rules#copyright-and-legality-of-content
[^3]: https://github.com/modrinth/labrinth/blob/master/src/search/indexing/local_import.rs#L68-L71
[^4]: https://api.modrinth.com/v2/tag/license

**Comment by Geometrically** [comment]
Could we have an "Advanced" option to have a list of common licenses that we do for the project creation? I think that would work and allow for some more robust-filtering abilities. It might be unnecessary though

**Comment by Lancelotbronner** [comment]
Thank you for the detailed answer, I didn't know Modrinth projects gave up that many rights when uploading, I guess that explains why a few authors refused to upload their mods.

I also believe we might be able then to do this on the client side, which should be enough for my needs.

Thank you!

**Comment by un-pogaz** [comment]
From what I know, these are the basic legal rights to allow Modrinth to host and share your mods. CurseForge, Planet Minecraft, or any other upload/sharing site, require the same thing somewhere in its EULA.
